internal/weldr/api.go

```diff
@@ -1940,22 +1955,31 @@ func ostreeResolveRef(location, ref string) (string, error) {
 	return parent, nil
 }
 
+// depsolveBlueprintForImageType handles depsolving the blueprint package list and
+// the packages required for the image type.
+// NOTE: The imageType *must* be from the same distribution as the blueprint.
 func (api *API) depsolveBlueprintForImageType(bp blueprint.Blueprint, imageType distro.ImageType) (map[string][]rpmmd.PackageSpec, error) {
 	// Depsolve using the host distro if none has been specified
 	if len(bp.Distro) == 0 {
 		bp.Distro = api.hostDistroName
 	}
 
+	if bp.Distro != imageType.Arch().Distro().Name() {
```

Comment:
I would honestly just panic here, this is a clear programming error.

Reply:
I don't like panic unless it's absolutely necessary, this should never happen but if it does I don't see anything wrong with returning an error.